### PR TITLE
Switch to serde_derive

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/13582)

### CODE REVIEW

**Comment by nox** [comment]
https://github.com/servo/rust-azure/pull/245
https://github.com/servo/webrender/pull/430

**Comment by nox** [comment]
https://github.com/serde-rs/syntex/pull/94

**Comment by nox** [comment]
https://github.com/rust-lang/cargo/issues/3164

**Review by Manishearth — APPROVED** [review]
r+

**Comment by nox** [comment]
The first commit fixes the bug discovered in https://github.com/dtolnay/syn/issues/31.

File: components/style/values/specified/mod.rs [hunk 1]
```diff
@@ -1469,16 +1469,16 @@ pub fn parse_border_width(input: &mut Parser) -> Result<Length, ()> {
 // The integer values here correspond to the border conflict resolution rules in CSS 2.1 §
 // 17.6.2.1. Higher values override lower values.
 define_numbered_css_keyword_enum! { BorderStyle:
-    "none" => none = -1,
```
Comment (Manishearth) [inline components/style/values/specified/mod.rs]: file a bug about undoing this

**Review by Manishearth — APPROVED** [review]
r+

File: components/style/values/specified/mod.rs (hunk 1 above)
Comment (nox) [inline components/style/values/specified/mod.rs]: Why would I do this? The negative values have no reason to exist AFAICT.

**Comment by nox** [comment]
According to @eddyb, this is fixed in a rustup. Closing this PR, will include the switch to serde_derive in the rustup.

**Comment by eddyb** [comment]
I didn't say it was. Only was curious if I was the cause of a new bug. If the context is macros 1.1 then it makes much more sense that the cause would be somewhere in there. Maybe it is fixed already on master, but idk for sure. cc @alexcrichton @nrc

**Comment by nox** [comment]
@eddyb Anyway, this is never going to merge without a rustup, and a rustup depends on this change too, so this PR is worthless on its own. :) Opened and tried https://github.com/servo/servo/pull/13649.

**Pull request closed without merging**